sdk_extensions/zpages/src/test/java/io/opentelemetry/sdk/extensions/zpages/TracezZPageHandlerTest.java

```diff
@@ -0,0 +1,289 @@
+        .contains("href=\"?zspanname=" + LATENCY_SPAN + "&ztype=1&zsubtype=8\">1");
+  }
+
+  @Test
+  public void summaryTable_linkForLatencyBasedSpans_MultipleForOneBoundary() {
+    OutputStream output = new ByteArrayOutputStream();
+    // 4 samples in boundary 5, >100ms
+    Span latencySpan100ms1 = tracer.spanBuilder(LATENCY_SPAN).setStartTimestamp(1L).startSpan();
+    EndSpanOptions endOptions1 = EndSpanOptions.builder().setEndTimestamp(112931232L).build();
+    latencySpan100ms1.end(endOptions1);
+    Span latencySpan100ms2 = tracer.spanBuilder(LATENCY_SPAN).setStartTimestamp(1L).startSpan();
+    EndSpanOptions endOptions2 = EndSpanOptions.builder().setEndTimestamp(138694322L).build();
+    latencySpan100ms2.end(endOptions2);
+    Span latencySpan100ms3 = tracer.spanBuilder(LATENCY_SPAN).setStartTimestamp(1L).startSpan();
+    EndSpanOptions endOptions3 = EndSpanOptions.builder().setEndTimestamp(154486482L).build();
+    latencySpan100ms3.end(endOptions3);
+    Span latencySpan100ms4 = tracer.spanBuilder(LATENCY_SPAN).setStartTimestamp(1L).startSpan();
+    EndSpanOptions endOptions4 = EndSpanOptions.builder().setEndTimestamp(194892582L).build();
+    latencySpan100ms4.end(endOptions4);
+
+    TracezZPageHandler tracezZPageHandler = new TracezZPageHandler(dataAggregator);
+    tracezZPageHandler.emitHtml(queryMap, output);
+
+    // Link for boundary 5 with 4 samples
+    assertThat(output.toString())
+        .contains("href=\"?zspanname=" + LATENCY_SPAN + "&ztype=1&zsubtype=5\">4");
+  }
+
+  @Test
+  public void summaryTable_linkForErrorSpans() {
+    OutputStream output = new ByteArrayOutputStream();
+    Span errorSpan1 = tracer.spanBuilder(ERROR_SPAN).startSpan();
+    Span errorSpan2 = tracer.spanBuilder(ERROR_SPAN).startSpan();
+    Span errorSpan3 = tracer.spanBuilder(ERROR_SPAN).startSpan();
+    Span finishedSpan = tracer.spanBuilder(FINISHED_SPAN_ONE).startSpan();
+    errorSpan1.setStatus(CanonicalCode.CANCELLED.toStatus());
+    errorSpan2.setStatus(CanonicalCode.ABORTED.toStatus());
+    errorSpan3.setStatus(CanonicalCode.DEADLINE_EXCEEDED.toStatus());
+    errorSpan1.end();
+    errorSpan2.end();
+    errorSpan3.end();
+    finishedSpan.end();
+
+    TracezZPageHandler tracezZPageHandler = new TracezZPageHandler(dataAggregator);
+    tracezZPageHandler.emitHtml(queryMap, output);
+
+    // Link for error based spans with 3 samples
+    assertThat(output.toString())
+        .contains("href=\"?zspanname=" + ERROR_SPAN + "&ztype=2&zsubtype=0\">3");
+    // No link for Status{#OK} spans
+    assertThat(output.toString())
+        .doesNotContain("href=\"?zspanname=" + FINISHED_SPAN_ONE + "&ztype=2&subtype=0\"");
+  }
+
+  @Test
+  public void spanDetails_emitSpanNameCorrectly() {
+    OutputStream output = new ByteArrayOutputStream();
+    Map<String, String> queryMapWithSpanName = new HashMap<String, String>();
+    queryMapWithSpanName.put("zspanname", FINISHED_SPAN_ONE);
+    queryMapWithSpanName.put("ztype", "1");
```

Comment:
I saw a third `.get` on subsampletype or something but see only two query parameters in this test so guessing we're missing one more test case.